src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -214,6 +198,39 @@ class IpexCommunicationService extends AgentService {
     return acdc;
   }
 
+  async getMatchingCredsForApply(notificationId: string) {
+    const keriNoti = await this.getNotificationRecordById(notificationId);
+    const msgSaid = keriNoti.a.d as string;
+    const msg = await this.signifyClient.exchanges().get(msgSaid);
+    const schemaSaid = msg.exn.a.s;
+    const attributes = msg.exn.a.a;
+    const creds = await this.signifyClient.credentials().list({
+      filter: {
+        "-s": { $eq: schemaSaid },
+        ...(Object.keys(attributes).length > 0
+          ? {
+            "-a": {
+              ...attributes,
+            },
+          }
+          : {}),
+      },
+    });
+    return Promise.all(
+      creds.map(async (cred: any) => {
+        const metadata = await this.credentialStorage.getCredentialMetadata(
+          `metadata:${cred.sad.d}`
+        );
+        if (!metadata) {
+          throw new Error(
+            IpexCommunicationService.CREDENTIAL_MISSING_METADATA_ERROR_MSG
+          );
+        }
+        return { ...getCredentialShortDetails(metadata), acdc: cred };
```

Comment:
This will need to be tidied up, not sure if we should throw an error if 1 metadata value is missing etc just in case. BUT, lets leave it and put a TODO

```
@TODO - This needs to be refined into a proper type once the UI has been decided. Also should potentially skip over missing credentials instead of throwing.
```